gen/calc.py

```diff
@@ -547,6 +547,53 @@ def validate_exhibitor_storage_master_discovery(master_discovery, exhibitor_stor
             "`master_http_load_balancer` then exhibitor_storage_backend must not be static."
 
 
+def calculate_adminrouter_tls_version_override(
+        adminrouter_tls_1_0_enabled,
+        adminrouter_tls_1_1_enabled,
+        adminrouter_tls_1_2_enabled):
+    tls_versions = list()
+    if adminrouter_tls_1_0_enabled == 'true':
+        tls_versions.append('TLSv1')
+
+    if adminrouter_tls_1_1_enabled == 'true':
+        tls_versions.append('TLSv1.1')
+
+    if adminrouter_tls_1_2_enabled == 'true':
+        tls_versions.append('TLSv1.2')
+
+    tls_version_string = " ".join(tls_versions)
+    return tls_version_string
+
+
+def calculate_adminrouter_tls_cipher_override(adminrouter_tls_cipher_suite):
```

Comment:
So we don't validate the cipher suite at all? When we get to the documentation, let's try to consider what could go wrong if I put "foo" in here.